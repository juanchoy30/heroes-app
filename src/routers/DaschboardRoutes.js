import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { Navbar } from '../components/ui/Navbar'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'

export const DaschboardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container mt-2">
                <Switch>
                    <Route exact path="/marvel" component={ MarvelScreen } />
                    <Route exact path="/hero/:heroId" component={ HeroScreen } />
                    <Route exact path="/dc/" component={ DcScreen } />
                    <Redirect to="/marvel" />
                </Switch>
            </div>

        </>
    )
}
