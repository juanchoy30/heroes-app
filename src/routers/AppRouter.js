import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DaschboardRoutes } from './DaschboardRoutes';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={ LoginScreen } />
                    <Route path="/" component={ DaschboardRoutes } />
                </Switch>
            </div>
        </Router>
    )
}
