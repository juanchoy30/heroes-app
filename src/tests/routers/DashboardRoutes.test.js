import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; // *1
import { configure, mount } from 'enzyme'; // *1
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';
import { MemoryRouter } from 'react-router-dom';

configure({adapter: new Adapter()}); 

// *1 This feature was not in the course. I found it on stackOverFlow, It seems to be
// a problem with the wojtekmaj adapter patch for enzima-react17. 

describe('<DashboardRoutes /> Tests', () => {

    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            user: 'Juan Miguel'
        }
    };
    
    test('should render correctly', () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contexValue }>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );
        expect( wrapper ).toMatchSnapshot();

    });
    

});
