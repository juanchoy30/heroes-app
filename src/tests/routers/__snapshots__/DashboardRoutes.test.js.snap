// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardRoutes /> Tests should render correctly 1`] = `ReactWrapper {}`;
